File: apps/agni_api/src/main/kotlin/dev/auguste/agni_api/infras/persistences/jbdc_model/AgentSuggestion.kt
package dev.auguste.agni_api.infras.persistences.jbdc_model

import dev.auguste.agni_api.core.entities.AgentSuggestion
import dev.auguste.agni_api.core.entities.enums.AgentSuggestionStatusType
import dev.auguste.agni_api.infras.persistences.IMapper
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import org.springframework.stereotype.Component
import java.util.UUID

@Table("agent_suggestions")
data class JdbcAgentSuggestionModel(
    @Id
    @get:JvmName("getIdentifier")
    @Column("agent_suggestion_id")
    val id: UUID,
    @Column("agent_id")
    val agentId: String,
    @Column("agent_name")
    val agentName: String,
    val title: String,
    val description: String,
    @Column("confidence_score")
    val confidenceScore: Double,
    val status: String
) : JdbcModel() {
    override fun getId(): UUID {
        return id
    }
}

@Component
class JdbcAgentSuggestionModelMapper: IMapper<JdbcAgentSuggestionModel, AgentSuggestion> {
    override fun toDomain(model: JdbcAgentSuggestionModel): AgentSuggestion {
        return AgentSuggestion(
            id = model.id,
            agentId = model.agentId,
            agentName = model.agentName,
            title = model.title,
            description = model.description,
            confidenceScore = model.confidenceScore,
            status = AgentSuggestionStatusType.fromString(model.status)
        )
    }

    override fun toModel(entity: AgentSuggestion): JdbcAgentSuggestionModel {
        return JdbcAgentSuggestionModel(
            id = entity.id,
            title = entity.title,
            agentId = entity.agentId,
            agentName = entity.agentName,
            description = entity.description,
            confidenceScore = entity.confidenceScore,
            status = entity.status.toString()
        )
    }

    override fun getSortField(): Set<String> {
        return setOf("agentId", "agentName", "confidenceScore", "status")
    }
}

File: apps/agni-web/mappers/agentSuggestion.ts
import type { ListResponse } from "~/types/api";
import type { GetAgentSuggestionResponse } from "~/types/api/agent-suggestion";
import type { AgentSuggestionStatusType, AgentSuggestionType } from "~/types/ui/agent-suggestion";

export function agentSuggestionResponseToAgentSuggestion(data: GetAgentSuggestionResponse): AgentSuggestionType {
    return {
        agentId: data.agentId,
        agentName: data.agentName,
        title: data.title,
        description: data.description,
        confidenceScore: data.confidenceScore,
        status: data.status as AgentSuggestionStatusType
    }
}

export function listAgentSuggestionsResponseToListAgentSuggestions(data: ListResponse<GetAgentSuggestionResponse>): ListResponse<AgentSuggestionType> {
    return {
        items: data.items.map(i => agentSuggestionResponseToAgentSuggestion(i)),
        total: data.total
    }
}
